### PR TITLE
PD-1308: Allow manual control of active group state for SideNavGroup

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

An issue came up in #leafygreen-ui about how an active SideNavItem that's wrapped by another component (such as a Tooltip or routing component) would not be detectable automatically by SideNavGroup. The addition of a `hasActiveItem` prop allows this state to be manually overwritten in cases where it may be incorrect.

🎟 _Jira ticket:_ [PD-1308](https://jira.mongodb.org/browse/PD-1308)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

## 🧪 How to test changes

1. Run storybook locally, and navigate to the "Simple Navigation" story under SideNav.
2. Toggle the `hasActiveItem` knob. You should see the state of the second item in the collapsed nav appear active or inactive based on the knob.

### CODE REVIEW

**Comment by brookescarlett** [comment]
Can we add an example to .design as well as getting a test or two in here?

File: packages/side-nav/src/SideNav.story.tsx [hunk 1]
```diff
@@ -250,6 +251,7 @@ function MockSideNav() {
           </SideNavGroup>
 
           <SideNavGroup
+            hasActiveItem={hasActiveItem}
```
Comment (brookescarlett) [inline packages/side-nav/src/SideNav.story.tsx:254]: Might just be me, but I'm not seeing anything change when I toggle this on/off

Comment (DesignerDave) [reply packages/side-nav/src/SideNav.story.tsx:254]: Is the nav collapsed? It only affects the collapsed nav.

File: packages/side-nav/README.md [hunk 2]
```diff
@@ -173,6 +173,7 @@ _Any other properties will be spread on the root-level element._
 | `children`         | `node`           | Content that will be rendered inside the root-level element.                                                                                     | `undefined` |
 | `collapsible`      | `boolean`        | Determines whether or not the Group is collapsible                                                                                               | `false`     |
 | `initialCollapsed` | `boolean`        | Determines whether or not the SideNavGroup should be collapsed on the first render.                                                              | `true`      |
+| `hasActiveItem`    | `boolean`        | Manually overrides automatic detection of whether a group contains an active item when the nav is collapsed.                                     |             |
```
Comment (brookescarlett) [inline packages/side-nav/README.md:176]: This language is confusing. Right now the only implication is styles applied when the nav is collapsed, but I don't think our implementation is limited to this (in the case that we want to use `isActiveGroup` for other logical decisions)

**Review by brookescarlett — APPROVED** [review]
My note still stands on I think we should remove "when collapsed" from the documentation, but I don't feel _that_ strongly. LGTM.

**Pull request merged**